### PR TITLE
Adapt reusable workflows for Rust apps

### PR DESCRIPTION
**Pull request by xchapron-ledger**

### CODE REVIEW

File: .github/workflows/_check_is_rust.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Clone app repository
+        uses: actions/checkout@v3
+        with:
+          path: app-repository
+          submodules: recursive
+
+      - name: Check presence of Rust app file indicator
+        id: get-is-rust
+        run: |
+          if [[ -e app-repository/ledger_app.toml ]];
```
Comment (agrojean-ledger) [inline .github/workflows/_check_is_rust.yaml:25]: I wonder if checking for `ledger_app.toml` file is enough to detect rust app. It's not mandatory to have this file is it ? For instance, https://github.com/LedgerHQ/app-avalanche does not have this file.

Comment (xchapron-ledger) [reply .github/workflows/_check_is_rust.yaml:25]: It's currently mandatory for an app hat need to be deployed as a Rust app.
App Avalanche is a C app which embed some Rust code.

File: scripts/cargo_metadata_dump.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.dirname(SCRIPT_DIR))
+
+from utils import run_cmd
+
+
+def save_app_params(device: str, app_build_path: Path, json_path: Path) -> None:
+    metadata = run_cmd("cargo metadata --no-deps --format-version 1 --offline -q",
+                       cwd=app_build_path)
+    metadata = json.loads(metadata)
```
Comment (yhql) [inline scripts/cargo_metadata_dump.py:18]: From Python you could use `toml` directly to extract the necessary info from the `Cargo.toml` (avoids a conversion to json)

File: .github/workflows/_check_clang_static_analyzer.yml
Comment (lpascal-ledger) [inline .github/workflows/_check_clang_static_analyzer.yml:1]: I guess we could rename the file and the workflow name to something like "Check code quality" or whatever as it's not longer clangSA-only.

File: .github/workflows/_check_is_rust.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+on:
+  workflow_call:
+    outputs:
+      is_rust:
+        description: Returns "true" if the app is using Rust SDK, else returns "false"
+        value: ${{ jobs.get-is-rust.outputs.is_rust }}
+
+jobs:
+  get-is-rust:
```
Comment (lpascal-ledger) [inline .github/workflows/_check_is_rust.yaml:11]: Why the sudden `-` when it's `_` everywhere else?

File: scripts/cargo_metadata_dump.py (hunk 2 above)
Comment (xchapron-ledger) [reply scripts/cargo_metadata_dump.py:18]: As discussed in direct, I'm keeping the usage of cargo metadata as this do'nt cost much and might allow detect weird conflict case in toml file which could be handled differently by cargo and python toml lib.

File: scripts/cargo_metadata_dump.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+from argparse import ArgumentParser
+from pathlib import Path
+import json
+import os
+import sys
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.dirname(SCRIPT_DIR))
```
Comment (lpascal-ledger) [inline scripts/cargo_metadata_dump.py:10]: Why is this needed? This script should be able to access local Python modules without needing to add its own directory into the `PYTHON_PATH`

File: scripts/utils.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python3
```
Comment (lpascal-ledger) [inline scripts/utils.py:1]: Superfluous. This file will never be executed as is.

File: scripts/makefile_dump.py [hunk 6]
```diff
@@ -1,11 +1,17 @@
...
 from argparse import ArgumentParser
 from pathlib import Path
 from typing import Tuple, List, Dict
 from tempfile import NamedTemporaryFile
 import json
+import os
+import sys
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.dirname(SCRIPT_DIR))
```
Comment (lpascal-ledger) [inline scripts/makefile_dump.py:12]: Same question. This surely is superfluous.

**Review by lpascal-ledger — APPROVED** [review]

File: .github/workflows/_check_clang_static_analyzer.yml
Comment (xchapron-ledger) [reply .github/workflows/_check_clang_static_analyzer.yml:1]: Yes but doing so will break CI on all repo as we currently enforce the job `_check_clang_static_analyzer`.
so I propose to rename it in a second time.

File: .github/workflows/_check_is_rust.yaml (hunk 3 above)
Comment (xchapron-ledger) [reply .github/workflows/_check_is_rust.yaml:11]: it's `-`on _get_workflow_version which I used as a template. But will change.
